pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/memory/DataBuffer.java

```diff
@@ -0,0 +1,292 @@
+   */
+  DataBuffer view(long start, long end, ByteOrder byteOrder);
+
+  /**
+   * Creates a view of the range [start, end) of this buffer with the current byte order. Calling {@link #flush()} or
+   * {@link #close()} has no effect on view.
+   */
+  default DataBuffer view(long start, long end) {
+    return view(start, end, order());
+  }
+
+  void flush();
+
+  default PinotInputStream openInputStream() {
+    return new DataBufferPinotInputStream(this);
+  }
+
+  default PinotInputStream openInputStream(long offset) {
+    return openInputStream(offset, size() - offset);
+  }
+
+  default PinotInputStream openInputStream(long offset, long length) {
+    return new DataBufferPinotInputStream(this, offset, offset + length);
+  }
+
+  /**
+   * Reads the range [offset, offset + length) as a RoaringBitmap.
+   * <p>
+   * Implementations should do their best to do not allocate memory and instead return a view of the underlying data.
+   */
+  ImmutableRoaringBitmap viewAsRoaringBitmap(long offset, int length);
+
+  /**
+   * Returns a ByteBuffer whose content is the same as the range [offset, offset + size) of this buffer.
+   * <p>
+   * Implementations should do their best to do not allocate memory and instead return a view of the underlying data.
+   * Callers cannot assume they have the ownership of the returned ByteBuffer, and therefore should not call
+   * {@link CleanerUtil#cleanQuietly(ByteBuffer)} using the returned object.
+   */
+  ByteBuffer copyOrView(long offset, int size, ByteOrder byteOrder);
+
+  void appendAsByteBuffers(List<ByteBuffer> appendTo);
+
+  /**
+   *
+   * Returns a ByteBuffer whose content is the same as the range [offset, offset + size) of this buffer.
+   * <p>
+   * Implementations should do their best to do not allocate memory and instead return a view of the underlying data.
+   * Callers cannot assume they have the ownership of the returned ByteBuffer, and therefore should not call
+   * {@link CleanerUtil#cleanQuietly(ByteBuffer)} using the returned object.
+   * <p>
+   * The returned ByteBuffer will have the same byte order as this buffer.
+   */
+  default ByteBuffer copyOrView(long offset, int size) {
+    return copyOrView(offset, size, order());
+  }
+
+  static boolean sameContent(DataBuffer buffer1, DataBuffer buffer2) {
+    if (buffer1.size() != buffer2.size()) {
+      return false;
+    }
+    DataBuffer nativeBuffer1 = buffer1.view(0, buffer1.size(), ByteOrder.nativeOrder());
+    DataBuffer nativeBuffer2 = buffer2.view(0, buffer1.size(), ByteOrder.nativeOrder());
+
+    long maxLong = nativeBuffer1.size() & ~7L;
+    for (long i = 0; i < maxLong; i += 8) {
+      if (nativeBuffer1.getLong(i) != nativeBuffer2.getLong(i)) {
+        return false;
+      }
+    }
+    for (long i = maxLong; i < buffer1.size(); i++) {
+      if (buffer1.getByte(i) != buffer2.getByte(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  static int commonHashCode(DataBuffer dataBuffer) {
```

Comment:
How are we planning to use this? Is this mainly used to avoid deserializing during comparison?